src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+                        case SyntaxKind.ForEachVariableStatement:
+                            return (int)SyntaxKind.ContinueStatement;
+                    }
+                }
+
+                return -1;
+            }
+
+            protected override (SyntaxNode first, SyntaxNode last) GetIfBodyStatementRange(IfStatementSyntax ifStatement)
+            {
+                var statement = ifStatement.Statement;
+                return (statement, statement);
+            }
+
+            protected override IEnumerable<(SyntaxNode first, SyntaxNode last)> GetSubsequentStatementRange(IfStatementSyntax ifStatement)
+            {
+                StatementSyntax innerStatement = ifStatement;
+                foreach (var node in ifStatement.Ancestors())
+                {
+                    var nextStatement = innerStatement.GetNextStatement();
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.Block:
+                            if (nextStatement != null) yield return (nextStatement, ((BlockSyntax)node).Statements.Last());
+                            break;
+                        case SyntaxKind.SwitchSection:
+                            if (nextStatement != null) yield return (nextStatement, ((SwitchSectionSyntax)node).Statements.Last());
+                            yield break;
+                        case SyntaxKind.LocalFunctionStatement:
+                        case SyntaxKind.SetAccessorDeclaration:
+                        case SyntaxKind.GetAccessorDeclaration:
+                        case SyntaxKind.AddAccessorDeclaration:
+                        case SyntaxKind.RemoveAccessorDeclaration:
+                        case SyntaxKind.MethodDeclaration:
+                        case SyntaxKind.ConstructorDeclaration:
+                        case SyntaxKind.DestructorDeclaration:
+                        case SyntaxKind.OperatorDeclaration:
+                        case SyntaxKind.ConversionOperatorDeclaration:
+                        case SyntaxKind.AnonymousMethodExpression:
+                        case SyntaxKind.SimpleLambdaExpression:
+                        case SyntaxKind.ParenthesizedLambdaExpression:
+                        case SyntaxKind.DoStatement:
+                        case SyntaxKind.WhileStatement:
+                        case SyntaxKind.ForStatement:
+                        case SyntaxKind.ForEachStatement:
+                        case SyntaxKind.ForEachVariableStatement:
+                            yield break;
+                    }
+
+                    if (node is StatementSyntax statement)
+                    {
+                        innerStatement = statement;
+                    }
+                }
+            }
+
+            protected override bool IsEmptyStatementRange((SyntaxNode first, SyntaxNode last) range)
+            {
+                return range.first == range.last && range.first.IsKind(SyntaxKind.EmptyStatement);
```

Comment:
curious if this is important enogh to have special handling for...

Reply:
Also, i don't recall tests that used empty statements.

Reply:
(pending)